### ISSUE TITLE
 Add official support for Python 3.13

### ISSUE THREAD

**Issue by astrojuanlu** [kedro-org/kedro#4267]
## Description
Just tested the Spaceflights starter on Python 3.13 and everything works smoothly.

See https://github.com/kedro-org/kedro/pull/3587 for the addition of official Python 3.12 support.

- [x] `kedro-org/kedro` #4328 
- [x] `kedro-org/kedro-viz` https://github.com/kedro-org/kedro-viz/pull/2190 https://github.com/kedro-org/kedro-viz/pull/2191
- [ ] `kedro-org/kedro-starters` https://github.com/kedro-org/kedro-starters/pull/258
- [ ] `kedro-org/kedro-plugins`
    - [ ] `kedro-telemetry` https://github.com/kedro-org/kedro-plugins/pull/936
    - [ ] `kedro-docker` https://github.com/kedro-org/kedro-plugins/pull/937
    - [ ] `kedro-datasets` https://github.com/kedro-org/kedro-plugins/pull/938
    - [ ] `kedro-airflow` https://github.com/kedro-org/kedro-plugins/pull/939
     
- [x] `kedro-org/vscode-kedro`

**Comment by astrojuanlu** [comment]
This will be the parent issue tracking all repos. The upgrade most likely needs to happen in a certain order.

- [ ] `kedro-org/kedro`
- [ ] `kedro-org/kedro-starters`
- [ ] `kedro-org/kedro-plugins`
- [ ] `kedro-org/kedro-viz`
- [ ] `kedro-org/vscode-kedro`

**Comment by noklam** [comment]
https://github.com/kedro-org/kedro/actions/runs/11666306391/job/32480967172?pr=4276

Adding some info here, seems like it breaks some micropkg command in 3.13 Windows

**Comment by ravi-kumar-pilla** [comment]
Hi Team,

There are some blockers adding python 3.13 support for `kedro-plugins`

1. **Kedro-airflow - blocked by apache-airflow support for python 3.13**
```ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (google-re2, msgspec)```
    Related issues as I could not find any active discussion in apache-airflow for python 3.13 -
      Google re2 : Open issues - https://github.com/google/re2/issues/516
      Msgspec : https://github.com/jcrist/msgspec/issues/764

2. **Kedro-datasets - blocked by tensorflow**

    Tensorflow - https://github.com/tensorflow/tensorflow/issues/78774

Thank you

**Comment by noklam** [comment]
Just to clarify, we can still do a release for `kedro` and `kedro-datasets` without python 3.13 support right? We break `kedro-datasets` exactly for this reason so I don't see that as a blocking factor.

**Comment by ravi-kumar-pilla** [comment]
> Just to clarify, we can still do a release for `kedro` and `kedro-datasets` without python 3.13 support right? We break `kedro-datasets` exactly for this reason so I don't see that as a blocking factor.

I did not fully understand what you meant by a release without python 3.13 support. So python 3.13 support is merged in Kedro but for kedro-datasets the build is failing as Tensorflow does not yet support python 3.13 and our [build](https://github.com/kedro-org/kedro-plugins/actions/runs/11846458385/job/33014136081?pr=932) depends on that.

**Comment by ravi-kumar-pilla** [comment]
Hi Team,

**Blockers for python 3.13 support on -**

**Kedro-plugins:**

- https://github.com/google/re2/issues/516
- https://github.com/jcrist/msgspec/issues/764
- https://github.com/tensorflow/tensorflow/issues/78774


**Kedro-starters:** The PR is merged but [waiting](https://github.com/python/cpython/issues/125235#issuecomment-2413365633) for 3.13.1 patch release

- https://github.com/python/cpython/issues/125235

Thank you

**Comment by merelcht** [comment]
Re `kedro-starters`: this is just for adding 3.13 builds to the starters, but doesn't make any changes in the code, so we're still good on releasing `kedro` with 3.13 support. It's also fine to wait with `kedro-airflow` and `kedro-datasets`. I suggest creating separate tickets for all of these so we can track them and the blockers individually.

**Comment by ravi-kumar-pilla** [comment]
Closing this in favor of - https://github.com/kedro-org/kedro-plugins/issues/940 and https://github.com/kedro-org/kedro-starters/issues/259

**Comment by noklam** [comment]
@ravi-kumar-pilla What I mean the dataset release should not be blocked by one dataset. We can temporarily skip that specific dataset for 3.13 until they release a compatible version
